### ISSUE TITLE
fix: 🤔 correct clear button color in form fields

### ISSUE THREAD

**Issue by coraliefeil** [solid-design-system/solid#1193]
## Current behavior

Button color for clear button is neutral-500 which has insufficient contrast on white and neutral-200 (hover) background.

## Expected behavior

Button color for clear button is neutral-800 which has sufficient contrast on white and neutral-200 (hover) background.

## Technical Information

- [x] Update & publish figma components sd-input, sd-input/sample-autosuggest, sd-select

- [ ] Update & merge storybook [sd-input](https://solid-design-system.fe.union-investment.de/x.x.x/storybook/?path=/story/components-sd-input--clearable), [sd-select](https://solid-design-system.fe.union-investment.de/x.x.x/storybook/?path=/story/components-sd-select--default&args=open-attr:!false;clearable-attr:!true;data-user-valid-attr:)

## DoR
- [x] Item has business value
- [x] Item has been estimated by the team
- [x] Item is clear and well-defined
- [x] Item dependencies have been identified

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch
